File: packages/dotnet/src/Models/Page.cs
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace Axene.Mailer
{
    /// <summary>
    /// A paginated envelope <c>{ items, total, page, limit }</c> returned by the
    /// suppressions list and webhook deliveries endpoints. <c>Page</c> is
    /// zero-based.
    /// </summary>
    /// <typeparam name="T">The item type.</typeparam>
    public sealed class Page<T>
    {
        /// <summary>The items on this page.</summary>
        [JsonPropertyName("items")] public List<T> Items { get; set; } = new List<T>();
        /// <summary>The total number of items across all pages.</summary>
        [JsonPropertyName("total")] public int Total { get; set; }
        /// <summary>The zero-based page index this envelope represents.</summary>
        [JsonPropertyName("page")] public int Page_ { get; set; }
        /// <summary>The page size.</summary>
        [JsonPropertyName("limit")] public int Limit { get; set; }
    }
}

File: packages/java/src/main/java/io/axene/mailer/Page.java
package io.axene.mailer;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import java.util.List;

/**
 * A paginated envelope {@code {items, total, page, limit}} returned by the
 * suppressions list and webhook-delivery list endpoints.
 *
 * @param <T> the element type.
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public final class Page<T> {
    /** The items on this page. */
    public List<T> items;
    /** Total number of items across all pages. */
    public int total;
    /** Zero-based page index. */
    public int page;
    /** Page size. */
    public int limit;
}
